### PR TITLE
API Integration and Artwork module

### PR DESCRIPTION
**Pull request by driftingly**
This PR adds the ability to add Artwork and associate artwork with prompts.

Artwork is added to JourneyMaker through the [AIC API](https://api.artic.edu/docs/). The API is queried by ID or reference number to populate the artwork module. The create modal uses a custom `QueryArtwork` vue component. Incorporating this custom component requires changes to the build step. `twill:build` is used to build Twill assets with custom Vue components/blocks.  This new build step adds more Twill assets to the `public` directory. Doing this build step as part of the deployment would allow us to remove these assets from source control. 

For some fields, the API will remain the source of truth. These fields appear as read-only on the frontend.

## WIP

Some areas of Artworks is still a WIP.

- Field validation, limit notes
- Improvements to Artwork listings
- Artworks added to the database seeder
- Cron for ensuring artworks stay up-to-date with API
- System for noting customized fields and changes to API data

### CODE REVIEW

**Review by nikhiltri — APPROVED** [review]
Wow, this is a ton of work! Looks great! ⭐

File: app/Support/Forms/Components/QueryArtwork.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        bool $translated = false,
+        bool $required = false,
+        ?string $note = '',
+        mixed $default = null,
+        bool $disabled = false,
+        bool $readOnly = false,
+        bool $inModal = false,
+        // Component specific
+        public ?string $placeholder = '',
+        public ?array $updateFormFields = [],
```
Comment (zachgarwood) [inline app/Support/Forms/Components/QueryArtwork.php:24]: I keep forgetting that we can do this now in PHP 8. 👍

File: database/migrations/2024_04_08_164938_create_artworks_tables.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        Schema::create('artworks', function (Blueprint $table) {
+            createDefaultTableFields($table);
+            $table->string('datahub_id');
+            $table->string('main_reference_number')->nullable();
+            $table->boolean('is_on_view')->nullable();
+            $table->string('credit_line')->nullable();
+            $table->string('copyright_notice')->nullable();
+            $table->decimal('latitude', 15, 13)->nullable();
+            $table->decimal('longitude', 15, 13)->nullable();
```
Comment (zachgarwood) [inline database/migrations/2024_04_08_164938_create_artworks_tables.php]: I made the same mistake with longitude in the mobile app CMS. Unlike latitude, which ranges from -90 to 90 degrees, longitude ranges from -180 to 180, so it needs an additional significant digit in its precision.

For reference, see my migration to fix this in the mobile CMS repo:
https://github.com/art-institute-of-chicago/aic-mobile-cms/blob/develop/database/migrations/2023_12_05_001409_correct_longitude_columns.php

**Review by zachgarwood — APPROVED** [review]
Nice as 🧊!

**Review by zachgarwood — CHANGES_REQUESTED** [review]
Small change to the longitude column in the migration

**Pull request merged**